app/routes.py

```diff
@@ -351,54 +361,105 @@ def add_harvest_source():
     )
 
 
-# View Source
+@mod.route("/harvest_source/<source_id>", methods=["GET"])
+def view_harvest_source_data(source_id: str):
+    source = db.get_harvest_source(source_id)
+    jobs = db.get_all_harvest_jobs_by_filter(
+        {"harvest_source_id": source.id, "status": "complete"}
+    )
+    next_job = "N/A"
+    future_jobs = db.get_new_harvest_jobs_by_source_in_future(source.id)
+    if len(future_jobs):
+        next_job = future_jobs[0].date_created
+    chartdata = {
+        "labels": [job.date_finished for job in jobs],
+        "datasets": [
+            {
+                "label": "Added",
+                "data": [job.records_added for job in jobs],
+                "borderColor": "green",
+                "backgroundColor": "green",
+            },
+            {
+                "label": "Deleted",
+                "data": [job.records_deleted for job in jobs],
+                "borderColor": "black",
+                "backgroundColor": "black",
+            },
+            {
+                "label": "Errored",
+                "data": [job.records_errored for job in jobs],
+                "borderColor": "red",
+                "backgroundColor": "red",
+            },
+            {
+                "label": "Ignored",
+                "data": [job.records_ignored for job in jobs],
+                "borderColor": "grey",
+                "backgroundColor": "grey",
+            },
+        ],
+    }
+    data = {
+        "harvest_source": source,
+        "harvest_source_dict": db._to_dict(source),
+        "harvest_jobs": jobs,
+        "chart": chartdata,
+        "next_job": next_job,
+    }
+    return render_template("view_source_data.html", data=data)
+
+
 @mod.route("/harvest_source/", methods=["GET"])
-@mod.route("/harvest_source/<source_id>", methods=["GET", "POST"])
-def get_harvest_source(source_id: str = None):
-    if source_id:
-        source = db._to_dict(db.get_harvest_source(source_id))
-        if request.args.get("type") and request.args.get("type") == "json":
-            return source
-        return render_template(
-            "view_data.html",
-            data=source,
-            action="View",
-            data_type="Harvest Source",
-            source_id=source_id,
-        )
-    else:
-        source = db.get_all_harvest_sources()
-        return db._to_dict(source)
+@mod.route("/harvest_sources/", methods=["GET"])
+def view_harvest_sources():
+    sources = db.get_all_harvest_sources()
+    data = {"harvest_sources": sources}
+    return render_template("view_source_list.html", data=data)
 
 
 # Edit Source
-@mod.route("/harvest_source/edit/<source_id>", methods=["GET", "POST"])
+@mod.route("/harvest_source/config/edit/<source_id>", methods=["GET", "POST"])
 @login_required
-def edit_harvest_source(source_id: str = None):
+def edit_harvest_source(source_id: str):
     if source_id:
```

Comment:
do we still need this condition if `source_id` is required?

Reply:
good point. probably not. i'll make the update.

Reply:
I'm going to defer this until the route cleanup ticket, if that's ok with you. There's some similar logic that exists in harvest_jobs and harvest_records that is inconsistent which should also be cleaned up.